go/nogo.rst

```diff
@@ -55,35 +57,32 @@ abstract syntax trees (ASTs) and type information for that package. For example:
       "github.com/bazelbuild/rules_go/go/tools/analysis"
     )
 
-    var Analysis = &analysis.Analysis{Name: "importunsafe", Run: run}
-
-    func init() {
-      analysis.Register(Analysis)
+    var Analyzer = &analysis.Analyzer{
+      Name: "importunsafe",
+      Doc: "reports imports of package unsafe",
+      Run: run,
     }
 
-    func run(p *analysis.Package) (*analysis.Result, error) {
+    func run(pass *analysis.Pass) (interface{}, error) {
       var findings []*analysis.Finding
-      for _, f := range p.Files {
+      for _, f := range pass.Files {
         for _, imp := range f.Imports {
           path, err := strconv.Unquote(imp.Path.Value)
           if err == nil && path == "unsafe" {
-            findings = append(findings, &analysis.Finding{
-              Pos:     imp.Pos(),
-              End:     imp.End(),
-              Message: "package unsafe must not be imported",
-            })
+            pass.Reportf(imp.Pos(), "package unsafe must not be imported")
           }
         }
       }
-      return &analysis.Result{Findings: findings}, nil
+      return nil, nil
     }
 
-Any findings returned by the check will fail compilation. Do not emit findings
-unless they are severe enough to warrant interrupting the compiler.
+Any diagnostics reported by the analyzer will fail compilation. Do not emit
+diagnostics unless they are severe enough to warrant interrupting the compiler.
 
-Each check must be written as a `go_tool_library`_ rule. This rule
-is identical to `go_library`_ but avoids a bootstrapping problem, which
-will be explained later. For example:
+Each analyzer must be written as a `go_tool_library`_ rule and must import a
+`go_tool_library`_ version of the package `analysis`_ target. This rule is
```

Comment:
Yeah, that message is good.

Users won't be able to use their own `golang.org/x/tools` unless they replace `@org_golang_x_tools` and use the same labels. For example, if they vendor `golang.org/x/tools`, they'll get build-time conflicts because the linker will see two different versions of `golang.org/x/tools/go/analysis.Analyzer` and other types.